android/src/main/java/com/guichaguri/trackplayer/service/player/LocalPlayback.java

```diff
@@ -106,10 +107,15 @@ public void remove(List<Integer> indexes, Promise promise) {
 
             if(i == 0) {
                 source.removeMediaSource(index, Utils.toRunnable(promise));
+                promiseWasCalled = true;
             } else {
                 source.removeMediaSource(index, null);
             }
         }
+        
+        if(!promiseWasCalled){
+            promise.resolve(null);
+        }
```

Comment:
It seems unnecessary adding a flag to check whether the promise was already resolved